cmd/argoexec/commands/wait.go

```diff
@@ -60,12 +60,14 @@ func waitContainer(ctx context.Context) error {
 	// Saving output parameters
 	err = wfExecutor.SaveParameters(ctx)
 	if err != nil {
+		wfExecutor.AnnotateOutputs(ctx, logArt)
```

Comment:
I think we need a slightly different approach here.

In each error branch of this func, we exit. Instead, we should just add `wfExecutor.AddError`. We do not need to exit until the end of the func, where we can exit with an error if there we any errors.